code/game/objects/items/devices/powersink.dm

```diff
@@ -141,8 +141,8 @@
 				if(istype(T.master, /obj/machinery/power/apc))

```

Comment:
needs a CHECK_TICK or STOP_LAG() before this line encase we start getting way too many nodes in the far future. This could technically run for a long ass time